modules/site-health/admin-site-health.php

```diff
@@ -0,0 +1,188 @@
+ *
+ * @since 2.8
+ * @link https://make.wordpress.org/core/2019/04/25/site-health-check-in-5-2/
+ */
+class PLL_Admin_Site_Health {
+	/**
+	 * PLL_Admin_Site_Health constructor.
+	 *
+	 * @since 2.8
+	 */
+	public function __construct() {
+		// Information tab
+		add_filter( 'debug_information', array( $this, 'pll_info_options' ), 15 );
+		add_filter( 'debug_information', array( $this, 'pll_info_languages' ), 16 );
+
+		// tests Tab
+		add_filter( 'site_status_tests', array( $this, 'pll_is_homepage' ) );
+	}
+
+	/**
+	 * Add Polylang Options to Site Health Informations tab.
+	 *
+	 * @param array $debug_info array options to display.
+	 * @return array
+	 * @since   2.8
+	 */
+	public function pll_info_options( $debug_info ) {
+		$options = get_option( 'polylang' );
+		$fields = array();
+		foreach ( $options as $key => $value ) {
+			if ( ! is_array( $value ) ) {
+				switch ( $key ) {
+					case 'first_activation':
+						$fields[ $key ]['label']   = $key;
+						$fields[ $key ]['value']   = date_i18n( 'd/m/Y', $value );
+						$fields[ $key ]['private'] = false;
+						break;
+					default:
+						$fields[ $key ]['label']   = $key;
+						$fields[ $key ]['value']   = $value;
+						$fields[ $key ]['private'] = false;
+						break;
+				}
+			}
+			if ( is_array( $value ) ) {
+				if ( empty( $value ) ) {
+					$fields[ $key ]['label']   = $key;
+					$fields[ $key ]['value']   = __( 'N/A', 'polylang' );
+					$fields[ $key ]['private'] = false;
+				} else {
+					$fields[ $key ]['label']   = $key;
+					$fields[ $key ]['value']   = implode( ', ', $value );
+					$fields[ $key ]['private'] = false;
+				}
+			}
+		}
+
+		$debug_info['polylang'] = array(
+			'label'    => __( 'Polylang Options', 'polylang' ),
+			'fields' => $fields,
+			'show_count' => true,
+		);
+
+		return $debug_info;
+	}
+
+	/**
+	 * Add Polylang Languages settings to Site Health Informations tab.
+	 *
+	 * @param array $debug_info array options to display.
+	 * @return array
+	 * @since   2.8
+	 */
+	public function pll_info_languages( $debug_info ) {
+		$languages = PLL()->model->get_languages_list();
+		$fields = array();
+		foreach ( $languages as $language ) {
+			foreach ( $language as $key => $value ) {
+				if ( empty( $value ) ) {
+					$value = __( 'N/A', 'polylang' );
+				}
+
+				$to_be_removed = array(
+					'flag' => true, // remove the flag as filter only display plain text
+					'host' => true, // Key not used by Polylang yet
+				);
+				if ( $to_be_removed[ $key ] ) {
+					continue;
+				}
+					$fields[ $key ]['label']   = $key;
+					$fields[ $key ]['value']   = $value;
+					$fields[ $key ]['private'] = false;
+			}
+			if ( empty( $language->flag ) ) {
+				$language->flag = __( 'Undefined', 'polylang' );
+			}
+			$debug_info[ $language->name ] = array(
+				'label'  => $language->name,
+				// translators: placeholder is the flag image
+				'description' => sprintf( __( 'Flag used in the language switcher: %s', 'polylang' ), $language->flag ),
```

Comment:
This may be not true. If the user has setup a custom flag, then the language switcher will display the custom flag, not this one.

Reply:
@Chouby is flag_url modified when a custom flag used ?

Reply:
Yes on frontend, no on admin. You can see the switch at: https://github.com/polylang/polylang/blob/2.7.4/include/model.php#L105-L110

Reply:
I don't remember if made that
```
/**
	 * Return the flag used in the language switcher
	 *
	 * @param object $language language object
	 * @return mixed
	 * @since   2.8
	 */
	public function get_flag( $language ) {
		if ( empty( $language->flag ) ) {
			$language->flag = __( 'Undefined', 'polylang' );
		}
		if ( ! empty( $language->custom_flag_url ) ) {
			$language->flag = '<img src="' . $language->custom_flag_url . '" height="11">';
		}
		return $language;
	}
``` to correct this comment on another but my custom flag is well displayed